File: lingting-core/src/main/java/live/lingting/framework/time/Time.java
package live.lingting.framework.time;

import live.lingting.framework.util.LocalDateTimeUtils;
import lombok.Getter;
import lombok.experimental.UtilityClass;

import java.time.LocalDateTime;
import java.time.temporal.TemporalAmount;
import java.time.temporal.TemporalUnit;

/**
 * @author lingting 2024-05-29 20:30
 */
@Getter
@UtilityClass
public class Time {

	/**
	 * 当前系统时间比实际真实时间慢多少毫秒
	 */
	private static long diff = 0;

	public static void setDiff(long diff) {
		Time.diff = diff;
	}

	public static long currentTimestamp() {
		return System.currentTimeMillis() + diff;
	}

	public static LocalDateTime current() {
		return LocalDateTimeUtils.parse(currentTimestamp());
	}

	public static LocalDateTime plus(TemporalAmount amount) {
		return current().plus(amount);
	}

	public static LocalDateTime plus(long amountToAdd, TemporalUnit unit) {return current().plus(amountToAdd, unit);}

	public static LocalDateTime plusYears(long years) {return current().plusYears(years);}

	public static LocalDateTime plusMonths(long months) {return current().plusMonths(months);}

	public static LocalDateTime plusWeeks(long weeks) {return current().plusWeeks(weeks);}

	public static LocalDateTime plusDays(long days) {return current().plusDays(days);}

	public static LocalDateTime plusHours(long hours) {return current().plusHours(hours);}

	public static LocalDateTime plusMinutes(long minutes) {return current().plusMinutes(minutes);}

	public static LocalDateTime plusSeconds(long seconds) {return current().plusSeconds(seconds);}

	public static LocalDateTime plusNanos(long nanos) {return current().plusNanos(nanos);}

	public static LocalDateTime minus(TemporalAmount amount) {
		return current().minus(amount);
	}

	public static LocalDateTime minus(long amountToAdd, TemporalUnit unit) {return current().minus(amountToAdd, unit);}

	public static LocalDateTime minusYears(long years) {return current().minusYears(years);}

	public static LocalDateTime minusMonths(long months) {return current().minusMonths(months);}

	public static LocalDateTime minusWeeks(long weeks) {return current().minusWeeks(weeks);}

	public static LocalDateTime minusDays(long days) {return current().minusDays(days);}

	public static LocalDateTime minusHours(long hours) {return current().minusHours(hours);}

	public static LocalDateTime minusMinutes(long minutes) {return current().minusMinutes(minutes);}

	public static LocalDateTime minusSeconds(long seconds) {return current().minusSeconds(seconds);}

	public static LocalDateTime minusNanos(long nanos) {return current().minusNanos(nanos);}

	public static String format() {
		return LocalDateTimeUtils.format(current());
	}

}

File: lingting-core/src/main/kotlin/live/lingting/framework/time/Time.kt
package live.lingting.framework.time

import java.time.LocalDateTime
import java.time.temporal.TemporalAmount
import java.time.temporal.TemporalUnit
import live.lingting.framework.util.LocalDateTimeUtils

/**
 * @author lingting 2024-05-29 20:30
 */
object Time {
    /**
     * 当前系统时间比实际真实时间慢多少毫秒
     */
    private var diff: Long = 0

    @JvmStatic
    fun setDiff(diff: Long) {
        this.diff = diff
    }

    @JvmStatic
    fun currentTimestamp(): Long {
        return System.currentTimeMillis() + diff
    }

    @JvmStatic
    fun current(): LocalDateTime {
        val timestamp = currentTimestamp()
        return LocalDateTimeUtils.parse(timestamp)
    }

    @JvmStatic
    fun plus(amount: TemporalAmount): LocalDateTime {
        return current().plus(amount)
    }

    @JvmStatic
    fun plus(amountToAdd: Long, unit: TemporalUnit): LocalDateTime {
        return current().plus(amountToAdd, unit)
    }

    @JvmStatic
    fun plusYears(years: Long): LocalDateTime {
        return current().plusYears(years)
    }

    @JvmStatic
    fun plusMonths(months: Long): LocalDateTime {
        return current().plusMonths(months)
    }

    @JvmStatic
    fun plusWeeks(weeks: Long): LocalDateTime {
        return current().plusWeeks(weeks)
    }

    @JvmStatic
    fun plusDays(days: Long): LocalDateTime {
        return current().plusDays(days)
    }

    @JvmStatic
    fun plusHours(hours: Long): LocalDateTime {
        return current().plusHours(hours)
    }

    @JvmStatic
    fun plusMinutes(minutes: Long): LocalDateTime {
        return current().plusMinutes(minutes)
    }

    @JvmStatic
    fun plusSeconds(seconds: Long): LocalDateTime {
        return current().plusSeconds(seconds)
    }

    @JvmStatic
    fun plusNanos(nanos: Long): LocalDateTime {
        return current().plusNanos(nanos)
    }

    @JvmStatic
    fun minus(amount: TemporalAmount): LocalDateTime {
        return current().minus(amount)
    }

    @JvmStatic
    fun minus(amountToAdd: Long, unit: TemporalUnit): LocalDateTime {
        return current().minus(amountToAdd, unit)
    }

    @JvmStatic
    fun minusYears(years: Long): LocalDateTime {
        return current().minusYears(years)
    }

    @JvmStatic
    fun minusMonths(months: Long): LocalDateTime {
        return current().minusMonths(months)
    }

    @JvmStatic
    fun minusWeeks(weeks: Long): LocalDateTime {
        return current().minusWeeks(weeks)
    }

    @JvmStatic
    fun minusDays(days: Long): LocalDateTime {
        return current().minusDays(days)
    }

    @JvmStatic
    fun minusHours(hours: Long): LocalDateTime {
        return current().minusHours(hours)
    }

    @JvmStatic
    fun minusMinutes(minutes: Long): LocalDateTime {
        return current().minusMinutes(minutes)
    }

    @JvmStatic
    fun minusSeconds(seconds: Long): LocalDateTime {
        return current().minusSeconds(seconds)
    }

    @JvmStatic
    fun minusNanos(nanos: Long): LocalDateTime {
        return current().minusNanos(nanos)
    }

    @JvmStatic
    fun format(): String {
        val current = current()
        return LocalDateTimeUtils.format(current)
    }
}

